application/src/main/java/org/togetherjava/tjbot/commands/BotCommandAdapter.java

```diff
@@ -54,17 +49,27 @@ protected BotCommandAdapter(CommandData data, CommandVisibility visibility) {
         this.data = data.setGuildOnly(visibility == CommandVisibility.GUILD);
         this.visibility = Objects.requireNonNull(visibility, "The visibility shouldn't be null");
 
-        this.name = data.getName();
-        this.type = data.getType();
+        type = commandTypeToInteractionType(data.getType());
+        componentIdInteractor = new ComponentIdInteractor(type, data.getName());
+    }
+
+    private static UserInteractionType commandTypeToInteractionType(Command.Type type) {
+        return switch (type) {
+            case SLASH -> UserInteractionType.SLASH_COMMAND;
+            case USER -> UserInteractionType.USER_CONTEXT_COMMAND;
+            case MESSAGE -> UserInteractionType.MESSAGE_CONTEXT_COMMAND;
+            case UNKNOWN -> throw new IllegalArgumentException(
+                    "Can not infer the type of user interaction for this command, it is unknown.");
+        };
```

Comment:
@Tais993 i intentionally did it like that, since its not meant to cover all `UserInteractionType`s, only the ones that `BotCommandAdapter` is interested in.

Like, `ComponentIdInteractor` wants a `UserInteractionType` and `BotCommandAdapter` has to provide one. In this particular case, the adapter decided to extract that out of the `CommandData`. So its its choice and job to do that.

On the other side, `UserInteractionType` can contain more enum entries that have nothing to do with this and are not compatible with that. Its not a 1:1 translation. In fact, it already contains `Other` which cant be mapped by this.